### ISSUE TITLE
bolt fails to deploy on hosts with debian bookworm

### ISSUE THREAD

**Issue by gucki** [bolt/bolt#7975]
Using Bolt 3.27.2.

Debian bookworm seems to have its own package of puppet-agent, which uses a different filesystem layout than the package provided by puppet directly. 

As bolt seems to have the path of the ruby interpreter hardcoded in some files, for example, custom_facts.rb, (`#! /opt/puppetlabs/puppet/bin/ruby`) the deployment fails: 

    $ bolt plan run deploy -t test01
    Starting: plan deploy
    Starting: install puppet and gather facts on test01
    Finished: plan deploy in 2.33 sec
    Failed on test01:
      The task failed with exit code 127 and no stdout, but stderr contained:
      bash: line 1: /tmp/d7de472f-aedc-4e88-bcb8-1e6b93b829af/custom_facts.rb: cannot execute: required file not found
    Failed on 1 target: test01
    Ran on 1 target

**Comment by bobdenotter** [comment]
Hi, you have the wrong Bolt.

**Comment by gucki** [comment]
@bobdenotter So it is already fixed in the lastest patch release 3.27.4 (only 2 patch versions above the one I'm using)?

**Comment by FredPraca** [comment]
It does not seem to, I just got this

```
$ bolt plan run server_install::configure -t test
Starting: plan server_install::configure
Starting: install puppet and gather facts on test
Finished: plan server_install::configure in 11.06 sec
Failed on test:
  The task failed with exit code 127 and no stdout, but stderr contained:
  bash: line 1: /tmp/fc36dad6-d110-4cd6-abb3-ed7f53cf2a07/custom_facts.rb: cannot execute: required file not found
Failed on 1 target: test
Ran on 1 target
```
with the right version...

```
$ bolt --version
3.27.4
```
